File: Week_01/G20200343030632/Leetcode_1_632.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = {}
        for i, num in enumerate(nums):
            if num in res:
                return res[num], i
            else:
               res[target - num] = i
        return None
            

# @lc code=end

File: Week_01/G20200343030632/Leetcode_1_632.java
import java.lang.Integer;
import java.util.Map;
import java.util.HashMap;
import java.util.Arrays;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> res = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (res.containsKey(nums[i])) {
                return new int[] {res.get(nums[i]), i};
            } else {
                res.put(target - nums[i], i);
            }
        }
        return null;
    }
    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 5};
        int target = 9;
        Solution obj = new Solution();
        int[] res =obj.twoSum(nums, target);
        System.out.println(Arrays.toString(res));


    }
}